modules/apps/social/social-activity-test-util/src/main/java/com/liferay/social/activity/test/util/BaseSocialActivityInterpreterTestCase.java

```diff
@@ -271,4 +192,109 @@ protected boolean isSupportsTrash() {
 	@Inject
 	protected TrashHelper trashHelper;
 
+	private void _checkInterpret() throws Exception {
+		List<SocialActivity> activities = getActivities();
+
+		Assert.assertFalse(activities.toString(), activities.isEmpty());
+
+		SocialActivityInterpreter activityInterpreter =
+			getActivityInterpreter();
+
+		for (SocialActivity activity : activities) {
+			if (hasClassName(activityInterpreter, activity.getClassName()) &&
+				hasActivityType(activity.getType())) {
+
+				SocialActivityFeedEntry activityFeedEntry =
+					activityInterpreter.interpret(activity, serviceContext);
+
+				Assert.assertNotNull(activityFeedEntry);
+
+				String title = activityFeedEntry.getTitle();
+
+				Assert.assertFalse(
+					"Title contains parameters: " + title,
+					title.matches("\\{\\d\\}"));
+			}
+		}
+	}
+
+	private void _checkLinks() throws Exception {
+		List<SocialActivity> activities = getActivities();
+
+		Assert.assertFalse(activities.toString(), activities.isEmpty());
+
+		SocialActivityInterpreter activityInterpreter =
+			getActivityInterpreter();
+
+		for (SocialActivity activity : activities) {
+			if (hasClassName(activityInterpreter, activity.getClassName()) &&
+				hasActivityType(activity.getType())) {
+
+				SocialActivityFeedEntry activityFeedEntry =
+					activityInterpreter.interpret(activity, serviceContext);
+
+				PortletURL portletURL = trashHelper.getViewContentURL(
+					serviceContext.getRequest(), activity.getClassName(),
+					activity.getClassPK());
+
+				if (Validator.isNull(activityFeedEntry.getLink()) &&
+					(portletURL == null)) {
+
+					continue;
+				}
+
+				Assert.assertEquals(
+					portletURL.toString(), activityFeedEntry.getLink());
+			}
+		}
+	}
+
+	private void _checkRenaming(List<SocialActivity> originalActivities)
+		throws Exception {
+
+		Assert.assertFalse(
+			originalActivities.toString(), originalActivities.isEmpty());
+
+		String originalTitle = _getActivitiesFirstTitle(originalActivities);
+		Set<Long> originalIds = _getActivitiesIds(originalActivities);
```

Comment:
We need to sort variable names alphabetically (unless there are dependencies between them).
```
Set<Long> originalIds = ...;
String originalTitle = ...;
```

Reply:
Also, as a general rule, it is better to fully spell the variable names; e.g. `originalIds` -> `originalActivitiesIds`.

Reply:
Wouldn't it be better to rename `originalTitle` to `firstActivityTitle`? Also, I think `_getActivitiesFirstTitle` should be renamed to `_getFirstActivityTitle` so that the name is more descriptive.

Reply:
Agree with all, except for `originalTitle` which gives an additional meaning to the variable. `_getFirstActivityTitle` is just the way we obtain this original title.